Assets/UI/Panels/productionpanel_CQUI.lua

```diff
@@ -214,22 +215,29 @@ function GetData()
     local buildQueue = pSelectedCity:GetBuildQueue();
 
     for row in GameInfo.Units() do
-        if row.MustPurchase and buildQueue:CanProduce( row.Hash, true ) and row.PurchaseYield == "YIELD_FAITH" then
+        if row.MustPurchase and buildQueue:CanProduce( row.Hash, true ) then
             local isCanProduceExclusion, results     = buildQueue:CanProduce( row.Hash, false, true );
             -- If a unit is purchase only, then "isDisabled" needs to be True, as that disables the button control that
             -- allows the religious units to be enqueued
-            -- TODO: It'd be nice to remove the unnecessary part about the production cost from the Tooltip string
             local isDisabled  :boolean = row.MustPurchase -- not isCanProduceExclusion;
             local sAllReasons :string = ComposeFailureReasonStrings( isDisabled, results );
             local sToolTip    :string = ToolTipHelper.GetUnitToolTip( row.Hash, MilitaryFormationTypes.STANDARD_MILITARY_FORMATION, buildQueue ) .. sAllReasons;
+            -- Remove the part about the production cost from the tooltip
+            local prodPartPattern = "%[NEWLINE%]%[NEWLINE%]Cost:.+%[ICON_Production%] Production"
+            sToolTip = sToolTip:gsub(prodPartPattern, "");
+
+            local turnsLeftIcon = CQUI_FAITH_ONLY_TURNS_LEFT;
+            if row.PurchaseYield == "YIELD_GOLD" then
+                turnsLeftIcon = CQUI_GOLD_ONLY_TURNS_LEFT;
+            end
 
             local kUnit :table = {
                 Type                = row.UnitType,
                 Name                = row.Name,
                 ToolTip             = sToolTip,
                 Hash                = row.Hash,
                 Kind                = row.Kind,
-                TurnsLeft           = CQUI_FAITH_ONLY_TURNS_LEFT,
```

Comment:
I don't get the logic here. Why turns left are used to mark purchase-only units? Using variables for other purpose than intended is bad programming.

Reply:
The TurnsLeft value is what is displayed in the production panel.  A good while back, a change was made to show the faith icon here instead of 999+, so this followed suit and would also show gold.
![image](https://user-images.githubusercontent.com/8787640/120983713-cefc5b00-c72e-11eb-8bb1-31035d159c5e.png)

GetTurnsToCompleteStrings subs in the icons if it sees the values that CQUI_FAITH_ONLY_TURNS_LEFT and CQUI_GOLD_ONLY_TURNS_LEFT map to (-2 and -3 respectively).  If I remember correct, the vertical misalignment of the costs values bothered me more than having [ICON_Faith][ICON_Turn] there.  For example:

![image](https://user-images.githubusercontent.com/8787640/120984491-8d1fe480-c72f-11eb-87c8-c4e0a92fd2d5.png)

The more I look at it, though, the more I like removing the "[ICON_Turn]" entirely and just letting it be vertically misaligned
![image](https://user-images.githubusercontent.com/8787640/120985328-5f876b00-c730-11eb-8193-fa55f1ab9db3.png)

![image](https://user-images.githubusercontent.com/8787640/120985321-5bf3e400-c730-11eb-8021-5421a9cfce40.png)

Reply:
the right fix is to muck with the productionpanel.xml if I really want to have those buttons always be vertically aligned, I think.  If I use a string consisting of 11 spaces, they get pretty close to aligned.  but I'll stick with what that last picture shows above.

Reply:
This is not a misalignment - these are different fields. One shows the price and the 2nd shows the turns. And turns are gone. The price is displayed correctly.

Reply:
Re: variable usage. I know that somone decided to use Turns Left to code "must purchase". That is why I said it was a bad programming. Anyway, this is not a big issue.